mmocr/utils/fileio.py

```diff
@@ -36,3 +40,66 @@ def list_from_file(filename, encoding='utf-8'):
         for line in f:
             item_list.append(line.rstrip('\n\r'))
     return item_list
+
+
+def is_archive(file_path: str) -> bool:
+    """Check whether the file is a supported archive format.
+
+    Args:
+        file_path (str): Path to the file.
+
+    Returns:
+        bool: Whether the file is an archive.
+    """
+
+    suffixes = ['zip', 'tar', 'tar.gz']
+
+    for suffix in suffixes:
+        if file_path.endswith(suffix):
+            return True
+    return False
+
+
+def check_integrity(file_path: str,
+                    md5: str,
+                    chunk_size: int = 1024 * 1024) -> bool:
+    """Check if the file exist and match to the given md5 code.
+
+    Args:
+        file_path (str): Path to the file.
+        md5 (str): MD5 to be matched.
+        chunk_size (int, optional): Chunk size. Defaults to 1024*1024.
+
+    Returns:
+        bool: Whether the md5 is matched.
+    """
+    if not osp.exists(file_path):
+        return False
+
+    if sys.version_info >= (3, 9):
+        hash = hashlib.md5(usedforsecurity=False)
+    else:
+        hash = hashlib.md5()
+    with open(file_path, 'rb') as f:
+        for chunk in iter(lambda: f.read(chunk_size), b''):
+            hash.update(chunk)
+
+    return hash.hexdigest() == md5
+
+
+def retrieve_files(path: str, suffixes: List) -> List:
```

Comment:
```suggestion
def list_files(path: str, suffixes: List) -> List:
```
retrieve_files sounds like it'll return file contents